### PR TITLE
[interop] Add grpc-java 1.58.0 to client_matrix.py

### PR DESCRIPTION
**Pull request by YifeiZhuang**

### CODE REVIEW

**Review by temawi — APPROVED** [review]

**Comment by temawi** [comment]
LGTM, but I don't have merge access to this repo so you still need approval from someone else.

**Comment by ejona86** [comment]
Can you share the link to the ad-hoc run?

**Comment by YifeiZhuang** [comment]
> Can you share the link to the ad-hoc run?

tap: https://fusion2.corp.google.com/ci/kokoro/prod:grpc%2Fcore%2Fexperimental%2Flinux%2Fgrpc_interop_matrix_adhoc/activity/66c7c93a-e58d-4714-96a7-893395c2c596/summary

**Review by ejona86 — APPROVED** [review]

**Pull request merged**
